src/mito2/src/request.rs

```diff
@@ -133,6 +134,17 @@ impl WriteRequest {
         })
     }
 
+    /// Returns estimated size of the request.
+    pub(crate) fn estimated_size(&self) -> usize {
+        let row_size = self
+            .rows
+            .rows
+            .get(0)
+            .map(|row| row.encoded_len())
```

Comment:
Question(Nonblocking): This is the length Tag-Length-Value (TLV) encoded data, there may be a significant difference in memory usage between the TLV data and actual data.

Reply:
Since we only compute an estimated size this should be enough since the `Rows` struct itself also occupies some spaces (size of struct and size of ColumnSchemas).